### DISCUSSION TITLE
Integrate LambdaTest test url links in spec reporter

### DISCUSSION THREAD

**Discussion by msr8830** [webdriverio/webdriverio#12324] [Ideas]
After testing this code locally, seem to work fine. Please add or share any feedback:

New getTestLink function:
`getTestLink({ sessionId, isMultiremote, instanceName, capabilities }) {
        const config = this.runnerStat && this.runnerStat.instanceOptions[sessionId];
        const isSauceJob = ((config && config.hostname && config.hostname.includes('saucelabs')) ||
            // only show if multiremote is not used
            capabilities && (
            // check w3c cap in jsonwp caps
            capabilities['sauce:options'] ||
                // check jsonwp caps
                capabilities.tunnelIdentifier ||
                // check w3c caps
                (capabilities.alwaysMatch &&
                    capabilities.alwaysMatch['sauce:options'])));
        const isLambdaTestJob = ((config && config.hostname && config.hostname.includes('lambdatest')) ||
          // only show if multiremote is not used
          capabilities && (
            // check w3c cap in jsonwp caps
            capabilities['LT:Options'] ||
            // check jsonwp caps
            capabilities.tunnelIdentifier ||
            // check w3c caps
            (capabilities.alwaysMatch &&
              capabilities.alwaysMatch['LT:Options'])));
        if ((isSauceJob || isLambdaTestJob) && config && config.user && config.key && sessionId) {
            const multiremoteNote = isMultiremote ? ` ${instanceName}` : '';
            const note = 'Check out%s job at %s';
            // The report url of RDC is in the caps that are returned
            if ('testobject_test_report_url' in capabilities) {
                return [format(note, multiremoteNote, capabilities.testobject_test_report_url)];
            }
            // VDC urls can be constructed / be made shared
            const isUSEast = config.headless || (config.hostname?.includes('us-east-1'));
            const isEUCentral = ['eu', 'eu-central-1'].includes(config?.region || '') || (config.hostname?.includes('eu-central'));
            const isAPAC = ['apac', 'apac-southeast-1'].includes(config?.region || '') || (config.hostname?.includes('apac'));
            const dc = isUSEast ? '.us-east-1' : isEUCentral ? '.eu-central-1' : isAPAC ? '.apac-southeast-1' : '';
            const sauceLabsSharableLinks = this._sauceLabsSharableLinks
                ? sauceAuthenticationToken(config.user, config.key, sessionId)
                : '';
            let remoteUrl;
            if (isSauceJob) {
              remoteUrl = `https://app${dc}.saucelabs.com/tests/${sessionId}${sauceLabsSharableLinks}`;
            } else if (isLambdaTestJob) {
              remoteUrl = `https://automation.lambdatest.com/logs/?sessionID=${sessionId}`;
            }
            return [format(note, multiremoteNote, remoteUrl)];
        }
        return [];
    }`

Old getTestLink function:
`getTestLink({ sessionId, isMultiremote, instanceName, capabilities }) {
        const config = this.runnerStat && this.runnerStat.instanceOptions[sessionId];
        const isSauceJob = ((config && config.hostname && config.hostname.includes('saucelabs')) ||
            // only show if multiremote is not used
            capabilities && (
            // check w3c cap in jsonwp caps
            capabilities['sauce:options'] ||
                // check jsonwp caps
                capabilities.tunnelIdentifier ||
                // check w3c caps
                (capabilities.alwaysMatch &&
                    capabilities.alwaysMatch['sauce:options'])));
        if (isSauceJob && config && config.user && config.key && sessionId) {
            const multiremoteNote = isMultiremote ? ` ${instanceName}` : '';
            const note = 'Check out%s job at %s';
            // The report url of RDC is in the caps that are returned
            if ('testobject_test_report_url' in capabilities) {
                return [format(note, multiremoteNote, capabilities.testobject_test_report_url)];
            }
            // VDC urls can be constructed / be made shared
            const isUSEast = config.headless || (config.hostname?.includes('us-east-1'));
            const isEUCentral = ['eu', 'eu-central-1'].includes(config?.region || '') || (config.hostname?.includes('eu-central'));
            const isAPAC = ['apac', 'apac-southeast-1'].includes(config?.region || '') || (config.hostname?.includes('apac'));
            const dc = isUSEast ? '.us-east-1' : isEUCentral ? '.eu-central-1' : isAPAC ? '.apac-southeast-1' : '';
            const sauceLabsSharableLinks = this._sauceLabsSharableLinks
                ? sauceAuthenticationToken(config.user, config.key, sessionId)
                : '';
            const sauceUrl = `https://app${dc}.saucelabs.com/tests/${sessionId}${sauceLabsSharableLinks}`;
            return [format(note, multiremoteNote, sauceUrl)];
        }
        return [];
    }`

<img width="1383" alt="image" src="https://github.com/webdriverio/webdriverio/assets/83895385/f0e4ad53-a56b-4fb9-a217-96ad3d134bfe">

**Comment by christian-bromann** [comment]
Sorry but I don't really understand what you are asking here. Please consider formatting your code and add more description thanks.

**Reply by msr8830** [reply]
Just realized, the format I used didn't turn out to be the way I wanted. When using sauceLabs we saw a link in the spec reporter: "Check out job at <Link to SauceLabs> .

We recently switched to LambdaTest and we no longer get that link, we used that link to look at tests that failed. So I added the LambdaTest link to the getTestLink function and it works for me. Wanted to see if the change I made makes sense and if its possible to implement this solution.

**Reply by msr8830** [reply]
The implemented solution works fine locally. I will try to reformat the code and provide a better representation.

**Reply by christian-bromann** [reply]
@msr8830 feel free to raise a pull requests with a link to the LambdaTest job for the spec reporter in `/packages/wdio-spec-reporter`. Please make sure to add proper test coverage for it. Thank you!
